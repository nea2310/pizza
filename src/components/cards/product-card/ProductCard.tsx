import { Link } from 'react-router-dom';
import './product-card.scss';
import { urlBuilder } from '../../../routes/routes';

type TProps = {
  image: string;
  isavl: boolean;
  name: string;
  pizzaid: string;
  price: number;
};

export default function (props: TProps) {

  let isAvl = null;
  if (!props.isavl) {
    isAvl = <span> Нет в наличии</span>;
  }
  return (
    <>
      <Link className='product-card__link'
        to={urlBuilder('productPage', { id: props.pizzaid })}>
        <h4
          className='product-card__header'>{props.name}</h4>
        <div className='product-card__img-wrapper'>
          <img
            className='product-card__img'
            src={props.image} alt="" />
        </div>

      </Link>
      <div className='product-card__info'>
        <strong className='product-card__price'>
          Цена: {props.price}р.</strong>
        <strong className='product-card__avl'>{isAvl}</strong>
      </div>
    </>
  );
}