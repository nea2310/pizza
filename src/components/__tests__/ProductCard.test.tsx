import { screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import ProductCard from '../cards/product-card/ProductCard';
import { renderWithProviders } from '../../shared/testUtils/testUtils';
import { initialStore } from '../../shared/testUtils/mockedStore';

describe('ProductCard component rendering', () => {
  it('should render Order button and Favorites button if user is signed in', () => {
    const productCardComponent = renderWithProviders(
      <BrowserRouter><ProductCard
        image={'http//'}
        isavl={true}
        name={'Четыре сыра'}
        pizzaid={'123'}
        price={100}
        incart={false}
        userdocid={'123456'}
        infav={false}
      /></BrowserRouter>, {
        preloadedState: initialStore
      }
    );
    expect(screen.getByText(/Четыре сыра/)).toBeInTheDocument();
    expect(screen.getByText(/Цена: 100р./)).toBeInTheDocument();
    expect(screen.getAllByRole('link')).toHaveLength(1);
    expect(screen.getAllByRole('button')).toHaveLength(2);
    expect(productCardComponent).toMatchSnapshot();
  });

  it('should render Sign in link and no Favorites button if user is not signed in', () => {
    const productCardComponent = renderWithProviders(
      <BrowserRouter><ProductCard
        image={'http//'}
        isavl={true}
        name={'Четыре сыра'}
        pizzaid={'123'}
        price={100}
        incart={false}
        userdocid={null}
        infav={false}
      /></BrowserRouter>, {
        preloadedState: initialStore
      }
    );
    expect(screen.getByText(/Четыре сыра/)).toBeInTheDocument();
    expect(screen.getByText(/Цена: 100р./)).toBeInTheDocument();
    expect(screen.getAllByRole('link')).toHaveLength(2);
    expect(screen.queryByRole('button')).not.toBeInTheDocument();
    expect(productCardComponent).toMatchSnapshot();
  });
});
