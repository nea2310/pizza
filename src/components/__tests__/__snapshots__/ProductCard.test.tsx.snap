// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component rendering should render Order button and Favorites button if user is signed in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="product-card"
      >
        <div
          class="product-card__favorites-button"
        >
          <button
            aria-label="добавить в избранное"
            class="favorites-button"
          />
        </div>
        <div
          class="product-card__content"
        >
          <h2
            class="product-card__header"
          >
            <a
              class="product-card__link"
              href="/product/123"
            >
              Четыре сыра
            </a>
          </h2>
          <img
            alt="Четыре сыра"
            class="product-card__img "
            src="http//"
          />
        </div>
        <div
          class="product-card__bottom"
        >
          <div
            class="product-card__info"
          >
            <span
              class="product-card__price"
            >
              Цена: 
              100
              р.
            </span>
          </div>
          <div
            class="product-card__order-button"
          >
            <button
              class="order-button order-button_type_order"
            >
              Заказать
            </button>
          </div>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="product-card"
    >
      <div
        class="product-card__favorites-button"
      >
        <button
          aria-label="добавить в избранное"
          class="favorites-button"
        />
      </div>
      <div
        class="product-card__content"
      >
        <h2
          class="product-card__header"
        >
          <a
            class="product-card__link"
            href="/product/123"
          >
            Четыре сыра
          </a>
        </h2>
        <img
          alt="Четыре сыра"
          class="product-card__img "
          src="http//"
        />
      </div>
      <div
        class="product-card__bottom"
      >
        <div
          class="product-card__info"
        >
          <span
            class="product-card__price"
          >
            Цена: 
            100
            р.
          </span>
        </div>
        <div
          class="product-card__order-button"
        >
          <button
            class="order-button order-button_type_order"
          >
            Заказать
          </button>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`ProductCard component rendering should render Sign in link and no Favorites button if user is not signed in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="product-card"
      >
        <div
          class="product-card__content"
        >
          <h2
            class="product-card__header"
          >
            <a
              class="product-card__link"
              href="/product/123"
            >
              Четыре сыра
            </a>
          </h2>
          <img
            alt="Четыре сыра"
            class="product-card__img "
            src="http//"
          />
        </div>
        <div
          class="product-card__bottom"
        >
          <div
            class="product-card__info"
          >
            <span
              class="product-card__price"
            >
              Цена: 
              100
              р.
            </span>
          </div>
          <div
            class="product-card__order-button"
          >
            <a
              class="product-card__login"
              href="/login"
            >
              Заказать
            </a>
          </div>
        </div>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="product-card"
    >
      <div
        class="product-card__content"
      >
        <h2
          class="product-card__header"
        >
          <a
            class="product-card__link"
            href="/product/123"
          >
            Четыре сыра
          </a>
        </h2>
        <img
          alt="Четыре сыра"
          class="product-card__img "
          src="http//"
        />
      </div>
      <div
        class="product-card__bottom"
      >
        <div
          class="product-card__info"
        >
          <span
            class="product-card__price"
          >
            Цена: 
            100
            р.
          </span>
        </div>
        <div
          class="product-card__order-button"
        >
          <a
            class="product-card__login"
            href="/login"
          >
            Заказать
          </a>
        </div>
      </div>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
